src/mpi/errhan/errutil.c

```diff
@@ -1024,16 +1024,21 @@ int MPIR_Err_create_code_valist(int lastcode, int fatal, const char fcname[],
                                 const char generic_msg[], const char specific_msg[], va_list Argp)
 {
     int generic_idx;
-    int errcode = lastcode;
+    int err_code;
     if (lastcode == MPI_SUCCESS) {
         generic_idx = FindGenericMsgIndex(generic_msg);
+        err_code = error_class;
         if (generic_idx >= 0) {
-            errcode = (generic_idx << ERROR_GENERIC_SHIFT) | error_class;
-            if (fatal)
-                errcode |= ERROR_FATAL_MASK;
+            err_code += (generic_idx + 1) << ERROR_GENERIC_SHIFT;
```

Comment:
Fixing a bug that we need generic_idx + 1

Reply:
@dalcinl The bug fix need be in separate commit; white space change in separate commit; spelling changes in separate commit; macro refactors in separate commit; actual changes in separate commit. I understand this may seem a hassle for contributors, but it helps the review *tremendously*, and ultimately avoids wasting contributor's time due to review delays.

Reply:
Don't forget that the ultimate time-saver for contributors is to not contribute. Unfortunately, next time I'll consider contribution I'll have to weigh in the benefits of providing cleanups and consolidations rather than just focusing in the minimal set of changes to get the fix I need accepted with no objections. 

This time, I consider your request to further split commits is unnecessary burden on me. The base, starting code I  had to deal with was old, there were bugs, had to review more modern code to consolidate style, come up with proper fixes while doing exhaustive testing. The whole thing was painful and unrewarding. And then I'm asked to roll back git history and go trough additional pain to further split commits. Such requests are reasonable in general, but in this particular instance, I honestly think it is unreasonable. Nonetheless, I will try do it. By engraving the pain in my brain, and I will learn my lesson.